### ISSUE TITLE
Smart Host Filter field "inventory" isn't working as expected

### ISSUE THREAD

**Issue by davidtluong** [ansible/awx#7201]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
When searching for inventory:example in Tower, it returns the following:

```
ERROR: Related Field got invalid lookup: icontains 
```

The below is the network debug:

```
Request URL:https://10.10.181.147/api/v2/hosts/?order_by=name&page_size=5&inventory__organization=1&host_filter=inventory__icontains=example
```

##### ENVIRONMENT
* Tower version: 3.6 and 3.7 
* AWX install method: Tower package installer
* Ansible version:  2.9.9
* Operating System: Fedora
* Web Browser: Firefox

##### STEPS TO REPRODUCE

Try to search for a key in smart host filter using `inventory:`

##### EXPECTED RESULTS

Returns the query successfully

##### ACTUAL RESULTS

```
ERROR: Related Field got invalid lookup: icontains 
```

##### ADDITIONAL INFORMATION

It looks like inventory is both in Related Fields and Fields for this window.  Also, searching for the inventory's id or name gives that same icontains error.

**Comment by Ben2064** [comment]
I'm running into the same issue on:
AWX 11.2
Chrome
Kubernetes (server)
Win 10 (client)

So it's probably not version/browser/os  related. For us this is making smart inventories essentially useless as we absolutely need to be able to filter by inventory/groups to make use of them in most cases.

**Comment by Toucan-Sam** [comment]
**TLDR:** inventory.name.icontains:"name of your inventory"

I'm also experiencing this and just making some observations.
AWX: 13.0.0 w/ Ansible 2.9.10

It looks like the 'inventory' field is both a 'field' and a 'related field'.

```
FIELDS: id, type, created, modified, name, description, inventory, enabled, instance_id, variables, last_job, last_job_host_summary, insights_system_id, ansible_facts_modified

RELATED FIELDS: created_by, jobs, groups, inventory, job_events_as_primary_host, ad_hoc_command_events, modified_by, last_job_host_summary, smart_inventories, last_job, job_host_summaries, ad_hoc_commands, inventory_sources, ansible_facts
```

`ERROR: Related Field got invalid lookup: icontains `

Might be related to icontains working on 'fields' but not 'related fields'? Appears to be the only duplicate named item. Either way the error message wasn't particularly clear about why I couldn't search for the thing.

Played with the api via `host.example/api/v2/` though and using `inventory.name.icontains:"name of your inventory"` seems to solve my search query problem. So not sure if bug, poorly documented, or just confusing in general.

**Comment by dwandro** [comment]
Have you tried using `inventory.name:"name of your inventory"`? This seems to work for me (using AWX 15.0.1)

**Comment by shanemcd** [comment]
Looks like this one auto-resolved itself. Please reopen if you see this in a newer version of AWX.
